src/main/java/org/testng/internal/Parameters.java

```diff
@@ -191,14 +189,15 @@
       annotationClass = ANNOTATION_MAP.get(annotation);
     }
 
-    return createParameters(m,
+    return createParameters(new ConstructorOrMethod(m),
         new MethodParameters(params,
             methodParams,
             parameterValues,
             currentTestMeth, ctx, testResult),
         finder, xmlSuite, annotationClass, name);
   }
 
+  @SuppressWarnings("unchecked")
```

Comment:
Maybe we can fix the warning instead:
`private static Class<? extends Annotation>[] annotationList = new Class<>[]`

Reply:
fixed.